aoc/d3/main.go

```diff
@@ -0,0 +1,72 @@
+package main
+
+import (
+	"fmt"
+	"os"
+	"strings"
+)
+
+// split a string in two using the middle point
+func SplitString(input string) []string {
+	half := len(input) / 2
+	return []string{input[:half], input[half:]}
+}
+
+// find the character that exists in two strings
+func FindCommonItem(input1 string, input2 string) string {
+	var result string
+	for _, item := range input1 { // generated by ChatGPT
+		if i := strings.Index(input2, string(item)); i != -1 {
+			result = string(item)
+		}
+	}
+	return result
+}
```

Comment:
edit: not casting, converting. Casting is different :)

Reply:
I like this a lot, thanks for the detailed explanation. The linter suggested used strings.ContainsRune instead of IndexRune and it works just fine. Added a test for exception handling too.
https://github.com/lucasrijllart/advent-of-code-2022/pull/1/commits/24ff2ac78c703702439b799d9687c1be9151342a